### PR TITLE
[docs] proposing header template for components

### PR DESCRIPTION
**Pull request by codeboten**
Testing out adding a new table to the top of the README for components. This header would give users information regarding:

- stability level
- supported pipelines
- distribution

Please provide feedback on what the best way to lay out this information is.

### CODE REVIEW

**Comment by TylerHelmuth** [comment]
I like the way this looks.  Should we link to the stability levels and/or the distributions?

File: processor/tailsamplingprocessor/README.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
-# Tail Sampling Processor
-
-Supported pipeline types: traces
+| Tail Sampling Processor  |               |
```
Comment (TylerHelmuth) [inline processor/tailsamplingprocessor/README.md:1]: I like keeping the processor title as a top-level header.  Could we get away without putting it in the table?

**Review by djaglowski — APPROVED** [review]
Nice improvement.

I'd slightly prefer to keep the component name as a `# Title`, but I don't have a better idea of what to put in the table header.

**Comment by codeboten** [comment]
> Nice improvement.
> 
> I'd slightly prefer to keep the component name as a `# Title`, but I don't have a better idea of what to put in the table header.

Yeah this is what I struggled with... github markdown requires something as a header... open to ideas

**Comment by codeboten** [comment]
> Should we link to the stability levels and/or the distributions?

Yes, I'll add this now

File: processor/tailsamplingprocessor/README.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
...
+| ------------------------ | ------------- |
+| Stability                | beta          |
+| Supported pipeline types | traces        |
+| Distribution             | core, contrib |
```
[the remark this answers is not in the document]
Comment (TylerHelmuth) [reply processor/tailsamplingprocessor/README.md:5]: Aren't there some components that live in contrib but are also in the core distro

[the remark this answers is not in the document]
Comment (codeboten) [reply processor/tailsamplingprocessor/README.md]: This is intended to show the distribution which contains the component, as some components are only in contrib, and not in core. Additionally some components are not yet contained in any of the supported distributions

[the remark this answers is not in the document]
Comment (codeboten) [reply processor/tailsamplingprocessor/README.md]: I've opened an issue to provide more information here: https://github.com/open-telemetry/opentelemetry-collector-releases/issues/121

File: processor/tailsamplingprocessor/README.md [hunk 3]
```diff
@@ -163,4 +167,7 @@ The probabilistic sampling policy makes decision based upon the trace ID, so wai
 You are already incurring the cost of running the tail sampling processor, adding the probabilistic policy will be negligible.
 Additionally, using the policy within the tail sampling processor will ensure traces that are sampled by other policies will not be dropped.
 
-[probabilistic_sampling_processor]: ../probabilisticsamplerprocessor
+[beta]: https://github.com/open-telemetry/opentelemetry-collector-contrib#beta
+[core]: https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol
+[contrib]: https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-contrib
```
Comment (TylerHelmuth) [inline processor/tailsamplingprocessor/README.md:172]: Can we link from the table?

File: processor/tailsamplingprocessor/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmitryax) [reply processor/tailsamplingprocessor/README.md]: Ok, it incorrect for tailsamplingprocessor then, should be `contrib` only, right?

File: processor/tailsamplingprocessor/README.md (hunk 3 above)
Comment (codeboten) [reply processor/tailsamplingprocessor/README.md:172]: These are linked in the table, i just defined the aliases for the links at the bottom

File: processor/tailsamplingprocessor/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (codeboten) [reply processor/tailsamplingprocessor/README.md]: The tail sampling processor has been in the core release the whole time. there are a bunch of components in the contrib repo that are part of the core release. the jaeger receiver/exporter is an example.

[the remark this answers is not in the document]
Comment (codeboten) [reply processor/tailsamplingprocessor/README.md]: Yup you are right.... apologies i thought the tail sampling processor was part of the core release. Will fix

File: processor/tailsamplingprocessor/README.md (hunk 1 above)
Comment (codeboten) [reply processor/tailsamplingprocessor/README.md]: Done

**Review by dmitryax — APPROVED** [review]

File: processor/tailsamplingprocessor/README.md [hunk 4]
```diff
@@ -1,6 +1,10 @@
 # Tail Sampling Processor
 
-Supported pipeline types: traces
+| Status                   |                   |
+| ------------------------ | ----------------- |
+| Stability                | [beta]            |
+| Supported pipeline types | traces            |
```
Comment (dmitryax) [inline processor/tailsamplingprocessor/README.md:6]:
```suggestion
| Status                   |           |
| ------------------------ | --------- |
| Stability                | [beta]    |
| Supported pipeline types | traces    |
```

**Review by dmitryax — APPROVED** [review]

File: processor/tailsamplingprocessor/README.md [hunk 5]
```diff
@@ -1,6 +1,10 @@
 # Tail Sampling Processor
 
-Supported pipeline types: traces
+| Status                   |           |
+| ------------------------ | --------- |
+| Stability                | [beta]    |
+| Supported pipeline types | traces    |
+| Distribution             | [contrib] |
```
Comment (bogdandrutu) [inline processor/tailsamplingprocessor/README.md:7]: s/Distribution/Distributions?

**Review by bogdandrutu — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**